ofono/bridge.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+from time import sleep
+
+import pydbus
+from dbus.mainloop.glib import DBusGMainLoop
+from gi.repository import GLib
+
+from helpers import Singleton, setup_logger
+
+DBusGMainLoop(set_as_default=True)  # has to be called first
+
+logger = setup_logger(__name__, 'debug')
+logging.basicConfig(level=logging.DEBUG)
+
+
+class OfonoBridge(object):
+    """
+    Generic util class to bridge between ZPUI and ofono backend through D-Bus
+    """
+
+    def __init__(self):
+        super(OfonoBridge, self).__init__()
+
+        bus = pydbus.SystemBus()
+        manager = bus.get('org.ofono', '/')
+        modem_path = manager.GetModems()[0][0]
+
+        if modem_path != '/sim900_0':
+            raise ValueError("Default modem should be '/sim900_0', was '{}'".format(modem_path))
+
+        # self.start()  #todo: check if it's the suitable place to do it
+
+    def start(self):
+        self.power_on()
+        self._init_messages()
+        self._listen_messages()
+
+    @property
+    def _bus(self):
+        # lil hack so we always have an up-to-date bus
```

Comment:
What's happening with the bus? Why can it get out-of-date?

Reply:
When you get the bus, you actually get a snapshot of the currently available methods. 

So if you get the bus before the modem is initialized you don't have any method. Using it that way ensures you always have the latest exposed methods.

Reply:
Understood! Would be great to put this explanation in the docstring/comment inside that function =)